io/net/net_tools.py

```diff
@@ -0,0 +1,289 @@
+        output = process.run(cmd="traceroute localhost",
+                                 shell=True, verbose=False)
+        if output.exit_status != 0:
+            self.error("\n traceroute failed")
+        no_of_hops = re.search(
+            "(\d+)\s+\S+\s*\(127.0.0.1\)",
+            output.stdout).group(1)
+        # Only one hop is required to get to localhost.
+        if (str(no_of_hops) != '1'):
+            self.error("traceroute did not show 1 hop for localhost")
+        if self.ipv6:
+            if self.detected_distro.name == "SuSE":
+                output = process.run(cmd="traceroute6 ipv6-localhost",
+                                         shell=True, verbose=False)
+            elif self.detected_distro.name == "Ubuntu":
+                output = process.run(cmd="traceroute6 ip6-localhost",
+                                         shell=True, verbose=False)
+            else:
+                output = process.run(cmd="traceroute6 localhost6",
+                                         shell=True, verbose=False)
+            if output.exit_status != 0:
+                self.error("\n traceroute6 failed")
+            no_of_hops = re.search(
+                "(\d+)\s+\S+\s*\(::1\)",
+                output.stdout).group(1)
+            if (str(no_of_hops) != '1'):
+                self.error("traceroute6 did not show 1 hop for localhost6")
+
+    def test_route(self):
+        """
+        Verify the route fuctionality.
+        """
+        output = process.run(cmd="route -n",
+                                 shell=True,
+                                 verbose=False)
+        if output.exit_status != 0:
+            self.error("route cmd failed")
+        if self.ipv6:
+            output = process.run(cmd="route -A inet6 -n",
+                                     shell=True,
+                                     verbose=False)
+            if output.exit_status != 0:
+                self.error("route ipv6 failed")
+
+    def test_ipmaddr(self):
+        """
+        Verify ipmaddr functionality
+        """
+        output = process.run(cmd="ipmaddr show dev lo",
+                                 shell=True,
+                                 verbose=False)
+        if output.exit_status != 0:
+            self.error("ipmaddr failed to diaplay mcast addr of loop back")
+        if self.ipv6:
+            output = process.run(cmd="ipmaddr show ipv6 dev lo",
+                                     shell=True,
+                                     verbose=False)
+            if output.exit_status != 0:
+                self.error("\n ipmaddr ipv6 failed")
+
+    def test_iptunnel(self):
+        """
+        Verify iptunnel functionality
+        """
+        # Note, this messes up dhcp client configurations so is skipped in that
+        # case
+        output = process.system_output("ps -aef", verbose=False, shell=True)
+        if 'dhclient' in output:
+            self.log.info("Skipped on dhclient systems")
```

Comment:
In that case it should be skipped. But you can only skip in the `setUp` phase... I'll finish the review and add some suggestions on this front...